src/Mvc/src/Microsoft.Extensions.ApiDescription.Design/GetFileReferenceMetadata.cs

```diff
@@ -17,6 +19,29 @@ public class GetFileReferenceMetadata : Task
     {
         private const string TypeScriptLanguageName = "TypeScript";
 
+        // Ignore Unicode escape sequences because, though they may escape valid class name characters,
+        // backslash is not a valid filename character.
+        private static readonly HashSet<UnicodeCategory> ValidClassNameCharacterCategories =
```

Comment:
Razor has a fairly straightforward implementation for producing a sanitized identifier. Could we copy-pasta the type just so that it's easier to patch them all if we were ever to patch one? https://github.com/aspnet/AspNetCore-Tooling/blob/master/src/Razor/src/Microsoft.AspNetCore.Razor.Language/CSharpIdentifier.cs

Reply:
Why does that class use such a limited selection of `UnicodeCategories`s (four)?

Reply:
Not super sure. Think @NTaylorMullen authored it

Reply:
Nope, lol those categories are from wayyyy back when. AspNetWebStack 😄

Reply:
If we incorporate the rest of the categories, I'd be willing to copy the code 😃

Reply:
Actually, I'd probably just reuse the updated class with the appropriate `<Compile />` item in the project.

Reply:
And, I'll file the appropriate bug sometime this evening.

Reply:
> Actually, I'd probably just reuse the updated class with the appropriate `<Compile />` item in the project.

I misread the URI and see `CSharpIdentifier` isn't in aspnet/AspNetCore. Will have to dupe the class ☹️

Reply:
Double-checked the `char.IsDigit(...)` and `char.IsLetter(...)` documentation. Nothing is missing given the broadness of those two calls. No issue on the AspNetCore-Tooling side.